docgen/src/guide/React_native.md

```diff
@@ -6,9 +6,643 @@ category: guide
+        return (
+          <Text key={idx} style={{ backgroundColor: '#ffff99' }}>
+            {part.value}
+          </Text>
+        );
+      return part.value;
+    });
+    return <Text>{highligtedHit}</Text>;
+  }
+);
+```
+
+Let’s use this new `<Highlight/>` component inside our `<Hits/>` template: 
+
+```jsx
+// [...]
+
+const Hits = connectInfiniteHits(({ hits, hasMore, refine }) => {
+  
+ // [...]
+
+  return (
+    <FlatList
+      data={hits}
+      onEndReached={onEndReached}
+      keyExtractor={keyExtractor}
+      renderItem={({ item }) => {
+        return (
+          <View style={{ flexDirection: 'row', alignItems: 'center' }}>
+            <Image
+              style={{ height: 100, width: 100 }}
+              source={{ uri: item.image }}
+            />
+            <View style={{ flex: 1 }}>
+              <Text>
+                <Highlight attributeName="name" hit={item} />
+
+              </Text>
+              <Text>
+                <Highlight attributeName="type" hit={item} />
+
+              </Text>
+            </View>
+          </View>
+        );
+      }}
+    />
+  );
+});
+```
+
+Now the search displays the results and highlights the part of the hit attribute that matches the query. This pattern is very important in search, especially with Algolia, so that the user knows what's going on. This way we create the search dialog between the user and the data.
+
+In this part, we've seen the following:
+- how to add a SearchBox to make queries into the records
+- how to highlight the matched part of the results
+- the importance of highlighting in a text-based search
+
+## Add RefinementList
+
+While the SearchBox is the way to go when it comes to textual search, you
+may also want to provide filters based on the structure of the records.
+
+Algolia provides a set of parameters for filtering by facets, numbers or geo
+location. Instantsearch packages those into a set of widgets and connectors.
+
+Since the dataset used here is an e-commerce one, let's add a RefinementList using the [connectRefinementList](connectors/connectRefinementList.html) connector
+to filter the products by categories.
+
+This connector gives you two interesting properties: 
+
+* `items `, the list of refinements. Each item has a `isRefined` property indicating if they are selected or not. 
+* `refine`, the function to call when a new category is selected
+
+On the React Native side we are going to take advantage of the [FlatList](https://facebook.github.io/react-native/docs/flatlist.html) and the [TouchableHighlight](http://facebook.github.io/react-native/releases/0.45/docs/touchablehighlight.html#touchablehighlight) component to render this refinement list. 
+
+```jsx
+// We need to add the RefinementList to our import
+import {
+  connectInfiniteHits,
+  connectSearchBox,
+  connectHighlight,
+  connectRefinementList,
+} from 'react-instantsearch/connectors';
+//We need to add the TouchableHighlight to our import
```

Comment:
Yes we can, I did not add it but that could be cool: https://github.com/eslint/eslint-plugin-markdown

Unsure it would reformat with prettier, to be tested